lib/bpk-component-text-link/src/BpkTextLink.ios.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import { TouchableWithoutFeedback } from 'react-native';
+import { primaryColor } from 'bpk-tokens/tokens/base.react.native';
+
+import {
+  BpkDynamicStyleSheet,
+  useBpkDynamicStyleSheet,
+} from '../../bpk-appearance';
+import BpkText from '../../bpk-component-text';
+import { getThemeAttributes, withTheme } from '../../bpk-theming';
+
+import { type Props, propTypes, defaultProps } from './common-types';
+import { REQUIRED_THEME_ATTRIBUTES } from './theming';
+
+const dynamicStyles = BpkDynamicStyleSheet.create({
+  text: {
+    color: primaryColor,
+  },
+});
+
+const BpkTextLink = (props: Props) => {
+  const { children, onPress, accessibilityLabel, theme, ...rest } = props;
+
+  const styles = useBpkDynamicStyleSheet(dynamicStyles);
+  const themeAttributes = getThemeAttributes(REQUIRED_THEME_ATTRIBUTES, theme);
+
+  const themeStyle = {
+    text: themeAttributes && {
+      color: themeAttributes.textLinkColor,
+    },
+  };
+
+  return (
+    <TouchableWithoutFeedback
+      accessibilityLabel={accessibilityLabel}
+      accessibilityRole="link"
+      onPress={onPress}
+    >
+      <BpkText style={[styles.text, themeStyle.text]} {...rest}>
```

Comment:
Wow would this work when it's inside a text with a different `textStyle` or `weight`? I believe the current implementation will not work as expected, it will show the link using the `base` text style even if it's inside a `lg` text, for example. We should probably add a story for it.

I think the work being done here is more or less what we need to make this work https://github.com/Skyscanner/backpack-react-native/pull/520

Reply:
I guess the idea here when we don't consider #520 is that consumers would pass the styles like they would for `BpkText` into this component, but I guess in the default case where users only apply the style to the parent `BpkText` it would be different as mentioned 😄

Reply:
If that is the case we need to extend the text props for this component as well. 

```js

type Props = $Exact<TextProps> & {
  <linkProps>
}
```

And the same for PropTypes

Reply:
Ok that sounds good to me - will make that change 😄